pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -602,6 +613,115 @@ func (pc *PodController) initializeSecondaryInterfaceStore() error {
 	return nil
 }
 
+// initializeSRIOVSecondaryInterfaceStore restores secondary interfaceStore for SR-IOV interfaces when agent restarts.
+// The source of truth for SR-IOV interfaces is the NetworkStatus annotation of Pods.
+func (pc *PodController) initializeSRIOVSecondaryInterfaceStore(primaryInterfaceStore interfacestore.InterfaceStore) error {
+	knownInterfaces := primaryInterfaceStore.GetInterfacesByType(interfacestore.ContainerInterface)
+	for _, ifconf := range knownInterfaces {
+		podNamespace := ifconf.ContainerInterfaceConfig.PodNamespace
+		podName := ifconf.ContainerInterfaceConfig.PodName
+		pod, err := pc.kubeClient.CoreV1().Pods(podNamespace).Get(context.TODO(), podName, metav1.GetOptions{})
+		if err != nil {
+			klog.ErrorS(err, "Failed to get Pod", "Pod", klog.KRef(podNamespace, podName))
+			continue
+		}
+		objRef := klog.KObj(pod)
+		_, found := checkForPodSecondaryNetworkAttachment(pod)
+		if !found {
+			klog.V(2).InfoS("Pod does not have a NetworkAttachmentDefinition", "Pod", objRef)
+			continue
+		}
+		netStatus, err := netdefutils.GetNetworkStatus(pod)
+		if err != nil {
+			klog.V(2).ErrorS(err, "Failed to get NetworkStatus for Pod", "Pod", objRef)
+			continue
+		}
+
+		cache, err := pc.buildVFDeviceIDListPerPod(podName, podNamespace)
```

Comment:
I do not understand how this func works well. Could you explain? How it fetches the device IDs for a Pod? Is it from Pod spec? Is it possible the information does not match the current state?

Reply:
Hi @jianjuns 

There is an assumption that there is only one device ID when the `buildVFDeviceIDListPerPod` for a corresponding `resource_name`. But it turns out it might be possible to have multiple device IDs, users can also request multiple VFs as well. I am going to refine the network status update part to include the pci_address as well, so that we can recover the map correct. Let me know what's your thoughts. Thanks.

Reply:
@jianjuns code is updated, for now, NetworkStatus will include pci address for VF device. Secondary interface restore will leverage the new added pci address from NetworkStatus and the resource_name from NAD to find matched resource in the cache (it will be built by buildVFDeviceIDListPerPod as long as the Pod is not deleted). 

Here we are putting in our best effort. One case we won't be able to handle is that Pod is force deleted when the agent is restarting. It should be accepted considering the impact is no VF name restoration but VF device can still be back to host network and reused.